### DISCUSSION TITLE
`intel_gpu_top` from within add-ons errors with `Failed to initialize PMU! (Permission denied)`

### DISCUSSION THREAD

**Discussion by felipecrs** [home-assistant/operating-system#2319] [General]
### Describe the issue you are experiencing

Frigate 0.12 has a feature to monitor GPU stats, and it uses `intel_gpu_top` for the task. It works normally when outside of the HaOS, but not inside of it. The add-on protection mode apparently makes no difference for the issue.

Looking up online I found some things:

> The command to make `intel_gpu_top` available for normal users:
> 
> ```shell
> setcap cap_perfmon=+ep /usr/bin/intel_gpu_top
> ```

https://gitlab.freedesktop.org/drm/intel/-/issues/5018

But neither setcap and intel_gpu_top is present in HassOS itself. Also, the add-on runs as root so I suspect that's not the issue.

Another thing that I found was https://gitlab.freedesktop.org/drm/igt-gpu-tools/-/issues/94, in which a maintainer asked:

> What's the kernel you're running? Does it have CONFIG_PERF_EVENTS enabled?

Which I don't the answer, but maybe HaOS's kernel lacks this _CONFIG_PERF_EVENTS_?

### What operating system image do you use?

generic-x86-64 (Generic UEFI capable x86-64 systems)

### What version of Home Assistant Operating System is installed?

9.4

### Did you upgrade the Operating System.

Yes

### Steps to reproduce the issue

1. Install Frigate 0.12 in an Intel based system (mine is a J4125)
1. Monitor the Frigate logs, you will see: `Failed to initialize PMU! (Permission denied)`

### Anything in the Supervisor logs that might be useful for us?

```txt
No.
```

### Anything in the Host logs that might be useful for us?

```txt
I don't think so, but here they are:

Jan 23 15:45:08 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.4Rqe8D.mount: Deactivated successfully.
Jan 23 15:45:31 homeassistant kernel: kauditd_printk_skb: 1 callbacks suppressed
Jan 23 15:45:31 homeassistant kernel: audit: type=1334 audit(1674488731.537:638): prog-id=0 op=UNLOAD
Jan 23 15:45:38 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.D2GSgP.mount: Deactivated successfully.
Jan 23 15:45:55 homeassistant systemd[1]: Starting Hostname Service...
Jan 23 15:45:55 homeassistant kernel: audit: type=1334 audit(1674488755.823:639): prog-id=161 op=LOAD
Jan 23 15:45:55 homeassistant kernel: audit: type=1334 audit(1674488755.823:640): prog-id=162 op=LOAD
Jan 23 15:45:55 homeassistant kernel: audit: type=1334 audit(1674488755.823:641): prog-id=163 op=LOAD
Jan 23 15:45:56 homeassistant systemd[1]: Started Hostname Service.
Jan 23 15:45:56 homeassistant kernel: audit: type=1334 audit(1674488756.195:642): prog-id=164 op=LOAD
Jan 23 15:45:56 homeassistant kernel: audit: type=1334 audit(1674488756.195:643): prog-id=165 op=LOAD
Jan 23 15:45:56 homeassistant kernel: audit: type=1334 audit(1674488756.195:644): prog-id=166 op=LOAD
Jan 23 15:45:56 homeassistant systemd[1]: Starting Time & Date Service...
Jan 23 15:45:56 homeassistant systemd[1]: Started Time & Date Service.
Jan 23 15:46:26 homeassistant systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Jan 23 15:46:26 homeassistant kernel: audit: type=1334 audit(1674488786.257:645): prog-id=0 op=UNLOAD
Jan 23 15:46:26 homeassistant kernel: audit: type=1334 audit(1674488786.257:646): prog-id=0 op=UNLOAD
Jan 23 15:46:26 homeassistant kernel: audit: type=1334 audit(1674488786.257:647): prog-id=0 op=UNLOAD
Jan 23 15:46:26 homeassistant systemd[1]: systemd-timedated.service: Deactivated successfully.
Jan 23 15:46:26 homeassistant kernel: audit: type=1334 audit(1674488786.586:648): prog-id=0 op=UNLOAD
Jan 23 15:46:26 homeassistant kernel: audit: type=1334 audit(1674488786.586:649): prog-id=0 op=UNLOAD
Jan 23 15:46:26 homeassistant kernel: audit: type=1334 audit(1674488786.586:650): prog-id=0 op=UNLOAD
Jan 23 15:46:38 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.d4CFDh.mount: Deactivated successfully.
Jan 23 15:47:09 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.UxNtw7.mount: Deactivated successfully.
Jan 23 15:47:39 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.i7Ez7G.mount: Deactivated successfully.
Jan 23 15:49:09 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.cOd1Ci.mount: Deactivated successfully.
Jan 23 15:49:54 homeassistant systemd[1]: docker-720cf19cbe40ba4c86513fc2174cad97f4dd3d97a1338e378fa1fadb7bc708e6.scope: Deactivated successfully.
Jan 23 15:49:54 homeassistant systemd[1]: docker-720cf19cbe40ba4c86513fc2174cad97f4dd3d97a1338e378fa1fadb7bc708e6.scope: Consumed 1min 9.157s CPU time.
Jan 23 15:49:54 homeassistant kernel: audit: type=1334 audit(1674488994.052:651): prog-id=0 op=UNLOAD
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.081603563Z" level=info msg="shim disconnected" id=720cf19cbe40ba4c86513fc2174cad97f4dd3d97a1338e378fa1fadb7bc708e6
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.081686776Z" level=warning msg="cleaning up after shim disconnected" id=720cf19cbe40ba4c86513fc2174cad97f4dd3d97a1338e378fa1fadb7bc708e6 namespace=moby
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.081709995Z" level=info msg="cleaning up dead shim"
Jan 23 15:49:54 homeassistant dockerd[429]: time="2023-01-23T15:49:54.082959550Z" level=info msg="ignoring event" container=720cf19cbe40ba4c86513fc2174cad97f4dd3d97a1338e378fa1fadb7bc708e6 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.124521748Z" level=warning msg="cleanup warnings time=\"2023-01-23T15:49:54Z\" level=info msg=\"starting signal loop\" namespace=moby pid=129528 runtime=io.containerd.runc.v2\n"
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.308406292Z" level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.308565197Z" level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.308595715Z" level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
Jan 23 15:49:54 homeassistant dockerd[438]: time="2023-01-23T15:49:54.311009471Z" level=info msg="starting signal loop" namespace=moby path=/run/docker/containerd/daemon/io.containerd.runtime.v2.task/moby/720cf19cbe40ba4c86513fc2174cad97f4dd3d97a1338e378fa1fadb7bc708e6 pid=129546 runtime=io.containerd.runc.v2
Jan 23 15:49:54 homeassistant systemd[1]: Started libcontainer container 720cf19cbe40ba4c86513fc2174cad97f4dd3d97a1338e378fa1fadb7bc708e6.
Jan 23 15:49:54 homeassistant kernel: audit: type=1334 audit(1674488994.506:652): prog-id=167 op=LOAD
Jan 23 15:49:54 homeassistant kernel: audit: type=1300 audit(1674488994.506:652): arch=c000003e syscall=321 success=yes exit=14 a0=5 a1=c000143a30 a2=78 a3=c0000249c0 items=0 ppid=129546 pid=129558 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
Jan 23 15:49:54 homeassistant kernel: audit: type=1327 audit(1674488994.506:652): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F6279002D2D6C6F67002F7661722F72756E2F646F636B65722F636F6E7461696E6572642F6461656D6F6E2F696F2E636F6E7461696E6572642E72756E74696D652E76322E7461736B2F6D6F62792F37323063663139
Jan 23 15:49:54 homeassistant kernel: audit: type=1334 audit(1674488994.506:653): prog-id=168 op=LOAD
Jan 23 15:49:54 homeassistant kernel: audit: type=1300 audit(1674488994.506:653): arch=c000003e syscall=321 success=yes exit=16 a0=5 a1=c0001437c8 a2=78 a3=c000024a08 items=0 ppid=129546 pid=129558 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="runc" exe="/usr/bin/runc" subj=unconfined key=(null)
Jan 23 15:49:54 homeassistant kernel: audit: type=1327 audit(1674488994.506:653): proctitle=72756E63002D2D726F6F74002F7661722F72756E2F646F636B65722F72756E74696D652D72756E632F6D6F6279002D2D6C6F67002F7661722F72756E2F646F636B65722F636F6E7461696E6572642F6461656D6F6E2F696F2E636F6E7461696E6572642E72756E74696D652E76322E7461736B2F6D6F62792F37323063663139
Jan 23 15:49:54 homeassistant kernel: audit: type=1334 audit(1674488994.506:654): prog-id=0 op=UNLOAD
Jan 23 15:49:54 homeassistant kernel: audit: type=1334 audit(1674488994.506:655): prog-id=0 op=UNLOAD
Jan 23 15:49:54 homeassistant kernel: audit: type=1334 audit(1674488994.506:656): prog-id=169 op=LOAD
Jan 23 15:50:10 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.UI7UUs.mount: Deactivated successfully.
Jan 23 15:50:40 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.4BkKrM.mount: Deactivated successfully.
Jan 23 15:51:40 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.a8Beih.mount: Deactivated successfully.
Jan 23 15:54:11 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.LASdDi.mount: Deactivated successfully.
Jan 23 15:57:12 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.iuWgri.mount: Deactivated successfully.
Jan 23 15:58:42 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.YmqfID.mount: Deactivated successfully.
Jan 23 16:00:13 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.SbtQGO.mount: Deactivated successfully.
Jan 23 16:02:13 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.9sPMO1.mount: Deactivated successfully.
Jan 23 16:02:43 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.oWlnNd.mount: Deactivated successfully.
Jan 23 16:03:14 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.lSayxD.mount: Deactivated successfully.
Jan 23 16:03:44 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.ERXcgC.mount: Deactivated successfully.
Jan 23 16:06:15 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.0InBce.mount: Deactivated successfully.
Jan 23 16:07:45 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.xrseGK.mount: Deactivated successfully.
Jan 23 16:08:45 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.21CqcG.mount: Deactivated successfully.
Jan 23 16:09:15 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.MtFB10.mount: Deactivated successfully.
Jan 23 16:09:46 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.vfmHGV.mount: Deactivated successfully.
Jan 23 16:10:16 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.lqdgaC.mount: Deactivated successfully.
Jan 23 16:10:46 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.8UBxkx.mount: Deactivated successfully.
Jan 23 16:12:47 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.N4t1Rh.mount: Deactivated successfully.
Jan 23 16:13:47 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.KjzDpk.mount: Deactivated successfully.
Jan 23 16:15:48 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.OMkejL.mount: Deactivated successfully.
Jan 23 16:16:49 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.MSujU1.mount: Deactivated successfully.
Jan 23 16:18:19 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.L4AGIK.mount: Deactivated successfully.
Jan 23 16:19:20 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.5b9X2n.mount: Deactivated successfully.
Jan 23 16:19:50 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.f5yag6.mount: Deactivated successfully.
Jan 23 16:20:20 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.9tBmZE.mount: Deactivated successfully.
Jan 23 16:21:51 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.AQGodF.mount: Deactivated successfully.
Jan 23 16:22:21 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.eLlxHE.mount: Deactivated successfully.
Jan 23 16:22:52 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.Z7WWtO.mount: Deactivated successfully.
Jan 23 16:23:52 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.p9Lyny.mount: Deactivated successfully.
Jan 23 16:25:53 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.lBjW9I.mount: Deactivated successfully.
Jan 23 16:26:23 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.axjfy3.mount: Deactivated successfully.
Jan 23 16:26:53 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.44J2CG.mount: Deactivated successfully.
Jan 23 16:27:24 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.YXYxor.mount: Deactivated successfully.
Jan 23 16:27:48 homeassistant dropbear[136584]: [136584] Jan 23 16:27:48 Child connection from 192.168.1.15:56540
Jan 23 16:27:49 homeassistant dropbear[136584]: [136584] Jan 23 16:27:49 Pubkey auth succeeded for 'root' with key sha1!! 35:84:96:b5:9f:37:50:82:c8:75:8d:12:3a:fa:ed:3a:15:31:88:30 from 192.168.1.15:56540
Jan 23 16:27:54 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.gOddY9.mount: Deactivated successfully.
Jan 23 16:28:24 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.kBpWlP.mount: Deactivated successfully.
Jan 23 16:28:54 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.gG0zgw.mount: Deactivated successfully.
Jan 23 16:29:55 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.qljW8D.mount: Deactivated successfully.
Jan 23 16:30:25 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.o6jSL9.mount: Deactivated successfully.
Jan 23 16:30:56 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.GfUJPs.mount: Deactivated successfully.
Jan 23 16:31:26 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.uTr9bB.mount: Deactivated successfully.
Jan 23 16:32:27 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.oCJvXQ.mount: Deactivated successfully.
Jan 23 16:33:57 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.hDmv4p.mount: Deactivated successfully.
Jan 23 16:35:58 homeassistant kernel: kauditd_printk_skb: 2 callbacks suppressed
Jan 23 16:35:58 homeassistant kernel: audit: type=1334 audit(1674491758.732:657): prog-id=170 op=LOAD
Jan 23 16:35:58 homeassistant systemd[1]: Started Journal Gateway Service.
Jan 23 16:35:58 homeassistant systemd-timesyncd[419]: Network configuration changed, trying to establish connection.
Jan 23 16:35:58 homeassistant systemd-timesyncd[419]: Network configuration changed, trying to establish connection.
Jan 23 16:35:58 homeassistant systemd-timesyncd[419]: Initial synchronization to time server 162.159.200.123:123 (time.cloudflare.com).
Jan 23 16:35:59 homeassistant systemd[1]: run-docker-runtime\x2drunc-moby-2209aee875eaaa3e67109bb1f975074416c33a0bf6b6825bc39176af0760d360-runc.V0Qwy8.mount: Deactivated successfully.
Jan 23 16:35:59 homeassistant systemd-journal-gatewayd[137925]: microhttpd: MHD_OPTION_EXTERNAL_LOGGER is not the first option specified for the daemon. Some messages may be printed by the standard MHD logger.
```

## System Information

version | core-2023.1.7
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.10.7
os_name | Linux
os_version | 5.15.80
arch | x86_64
timezone | America/Sao_Paulo
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4980
Installed Version | 1.30.0
Stage | running
Available Repositories | 1204
Downloaded Repositories | 12
HACS Data | ok

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 9.4
-- | --
update_channel | stable
supervisor_version | supervisor-2022.12.1
agent_version | 1.4.1
docker_version | 20.10.19
disk_total | 234.0 GB
disk_used | 65.6 GB
healthy | true
supported | true
board | generic-x86-64
supervisor_api | ok
version_api | ok
installed_addons | Studio Code Server (5.5.2), SSH & Web Terminal (13.0.2), Mosquitto broker (6.1.3), Cloudflared (4.0.8), Asterisk (2.3.5), RPC Shutdown (2.2), ESPHome (2022.12.5), AdGuard Home (4.8.0), Glances (0.17.2), Log Viewer (0.14.0), OneDrive Backup (1.4), LibreTuya ESPHome (latest), MariaDB (2.5.1), Nginx Proxy Manager (0.12.3), Traccar (0.17.0), Frigate Beta (0.12.0) (0.12.0-beta5)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 6
views | 4
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | January 15, 2023 at 3:46 AM
-- | --
current_recorder_run | January 23, 2023 at 12:49 PM
estimated_db_size | 1000.78 MiB
database_engine | sqlite
database_version | 3.38.5

</details>

<details><summary>Sonoff</summary>

version | 3.3.1 (b20e33c)
-- | --
cloud_online | 2 / 2
local_online | 2 / 2

</details>

### Additional information

Refs https://github.com/blakeblackshear/frigate/discussions/5156#discussioncomment-4758233 and https://github.com/blakeblackshear/frigate/issues/3912.

**Comment by felipecrs** [comment]
May I ask why this was converted from issue to discussion? I opened it as being a bug report.

**Reply by agners** [reply]
It has been converted to a discussion because it is not a bug in HAOS. Sorry, I got distracted while writing my comment, I am finishing up my answer right now.

**Comment by agners** [comment]
> > The command to make `intel_gpu_top` available for normal users:
> > ```shell
> > setcap cap_perfmon=+ep /usr/bin/intel_gpu_top
> > ```
> 
> https://gitlab.freedesktop.org/drm/intel/-/issues/5018
> 
> But neither setcap and intel_gpu_top is present in HassOS itself. Also, the add-on runs as root so I suspect that's not the issue.

This are actually very good hints. Nowadays, root != root: Just because a process is running as root (uid 0) does not mean it has all rights. A root (privileged) user has [capability](https://man7.org/linux/man-pages/man7/capabilities.7.html) flags. A full root has all capabilities. Typically root inside a containers has very few flags set (and that is how the add-on runs, as a container).

You can configure what capabilities your add-on requires by setting the `privileged` property in the [add-on configuration](https://developers.home-assistant.io/docs/add-ons/configuration) properly. It seems that intel_gpu_top requires `CAP_PERFMON`. However, I think that flag is (not yet) supported by Supervisor: https://github.com/home-assistant/supervisor/blob/ee1daa0b353aedcb689fdc464bc3c7500e77a4a3/supervisor/docker/const.py#L5-L17. You can try adding it to the add-on configuration and see what happens. If it doesn't work, you have to adjust the Supervisor. 

> Which I don't the answer, but maybe HaOS's kernel lacks this CONFIG_PERF_EVENTS?

The kernel config seems to be enabled (`CONFIG_PERF_EVENTS=y`, at least for generic-x86-64).

**Reply by felipecrs** [reply]
Thanks for the answer!

I guess that by `privileged` you mean to add [`full_access: true` in the add-on `config.yaml`](https://github.com/blakeblackshear/frigate-hass-addons/blob/691f2be2c13655c0bfc40e6d2c87c65b4e42d3c0/frigate_fa_beta/config.yaml#L36), and after installing it, disable the _Protection Mode_, right?

Then, yes. An user has reported that the same issue occurs even with that: https://github.com/blakeblackshear/frigate/issues/3912#issuecomment-1280785575

So, I'd say that chances are that this `CAP_PERFMON` capability is missing indeed in Supervisor. What's your suggestion? I can send a PR to add it to the list, it seems easy. :)

**Reply by felipecrs** [reply]
Oh, sorry, I just realized what you meant now and it's not full_access. Ok, I'll give it a try. Thanks.

**Reply by felipecrs** [reply]
Ok, I tried here, this way:

![image](https://user-images.githubusercontent.com/29582865/214687874-6936701a-ccc0-47ea-b5f9-ab0e7699e8f6.png)

But as soon as I add this, the add-on disappears from the add-ons list in the store. I suppose there should be some validation of the fields, and it rejects it because it's not one of the allowed values.

So, back to my first comment. Any suggestions on the way forward?

**Reply by felipecrs** [reply]
Ok... actually it seems that it's not yet available in Docker, will be only available in 22.x:

> 
> Current versions of containerd and runc support these, so possibly we can
> include them again in a patch release, but otherwise support will come in
> the next (22.06) release of docker engine.

https://github.com/docker/docs/pull/15050

**Reply by felipecrs** [reply]
Actually I don't know. This is confusing!

**Reply by agners** [reply]
Yeah support for this capability seems to be missing currently in Docker :cry: 

```
$ docker --version
Docker version 20.10.23, build 715524332f
$ docker run -it --cap-add=CAP_PERFMON debian /bin/bash 
docker: Error response from daemon: invalid CapAdd: unknown capability: "CAP_PERFMON".
```

**Reply by agners** [reply]
> But as soon as I add this, the add-on disappears from the add-ons list in the store. I suppose there should be some validation of the fields, and it rejects it because it's not one of the allowed values.
> 
> So, back to my first comment. Any suggestions on the way forward?

Could also be that Supervisor does a validation, after all there is a list of capabilities in the Supervisor code. I guess the capability needs to be added there. But as long as Docker is not there, not much we can do :cry:

**Reply by agners** [reply]
I mean, there is a way to do it:

1. Patch Docker/runc in Home Assistant OS to support `CAP_PERFMON`.
2. Extend Supervisor to support `CAP_PERFMON` if OS == HAOS (version x.y), otherwise ignore it
3. Extend Frigate add-on to check if `CAP_PERFMON` is set, if so use it, if not inform the user (for cases where we don't control Docker, e.g. Supervised).

**Reply by felipecrs** [reply]
Yeah... Not worth doing, I think. This is not a mandatory feature anyway.

Well, thanks a lot for helping figure this out. I'll keep an eye on it, when it's finally released in docker, I'll follow up to have support added in supervisor.

**Reply by felipecrs** [reply]
Docker 23.0.0 is out now, and it supports CAP_PERFMON.

https://github.com/moby/moby/releases/tag/v23.0.0

**Reply by agners** [reply]
Hm, really fresh, but doesn't look like there are blocking breaking changes, at first sight. Maybe worth considering for HAOS 10.

**Reply by felipecrs** [reply]
- https://github.com/home-assistant/buildroot/pull/5
- https://github.com/home-assistant/supervisor/pull/4130
- https://github.com/home-assistant/developers.home-assistant/pull/1661

**Comment by simyz450f2** [comment]
git clone version: 0.12.0-beta7 /addons

add to config.yaml

name: Frigate (Full Access) Beta (0.12.0)
version: 0.12.0-beta7
panel_icon: "mdi:cctv"
panel_title: Frigate
slug: frigate-fa-beta
description: NVR with realtime local object detection for IP cameras
url: "https://github.com/blakeblackshear/frigate"
image: ghcr.io/blakeblackshear/frigate
startup: application
boot: auto
init: false
webui: "http://[HOST]:[PORT:5000]/"
watchdog: "http://[HOST]:[PORT:5000]/"
ingress: true
ingress_port: 5000
ingress_entry: /
panel_admin: false
homeassistant_api: true
hassio_api: true
ports:
  8555/tcp: 8555
  8555/udp: 8555
  8554/tcp: 8554
  5000/tcp: 5000
  1984/tcp: 1984
  1935/tcp: 1935
ports_description:
  8555/tcp: WebRTC over tcp
  8555/udp: WebRTC over udp
  8554/tcp: RTSP Restream
  5000/tcp: Web interface (Not required for Hass.io Ingress)
  1984/tcp: go2rtc API
  1935/tcp: RTMP streams
host_network: false
tmpfs: true
full_access: true
privileged: 
  - SYS_ADMIN
environment:
  CONFIG_FILE: /config/frigate.yml
schema:
  plus_api_key: password?
services:
  - "mqtt:need"
arch:
  - amd64
  - aarch64
  - armv7
map:
  - "media:rw"
  - "config:rw"

refresh addon repos
install 
enjoy

**Comment by mew1033** [comment]
I just figured out how to get this working without privileged mode. You need the `CAP_PERFMON` capability like before, so only newer docker. You also need to check `/proc/sys/kernel/perf_event_paranoid` and set it to <=2. https://github.com/blakeblackshear/frigate/pull/6166

**Reply by agners** [reply]
HAOS uses `perf_event_paranoid=2` already, so we should be good on this front. Since we use Docker 23.0 now, I guess the only thing remaining is adding support to Supervisor then?

**Reply by agners** [reply]
> I guess the only thing remaining is adding support to Supervisor then?

Ah I see this is already ongoing with https://github.com/home-assistant/supervisor/pull/4259.
